autogluon_zeroshot/simulation/simulation_context.py

```diff
@@ -213,7 +215,7 @@ def load_pred(self, pred_pkl_path: Path, lazy_format: bool = False) -> dict:
 
     @staticmethod
     def convert_lazy_format(pred_pkl_path, override_if_already_exists: bool = False):
-        new_filename = pred_pkl_path.parent / "zeroshot_pred_per_task"
+        new_filename = Path(str(pred_pkl_path).rsplit('.pkl', 1)[0])
```

Comment:
Why is this change needed? If it is needed, can we use Path rather to avoid potential portability issue? (eg `Path(pred_pkl_path).parent / Path(pred_pkl_path).stem`)

Reply:
I sometimes have suffixes to the names of the pkl files, so this allows flexibility to simply have the folder be the name of the file rather than a fixed `zeroshot_pred_per_task`. Otherwise they would overwrite eachother.

```
    "results/bagged_289/zeroshot_pred_proba_10_mb.pkl": "s3://automl-benchmark-ag/aggregated/ec2/2023_03_19_zs/zeroshot_pred_proba_10_mb.pkl",
    "results/bagged_289/zeroshot_pred_proba_50_mb.pkl": "s3://automl-benchmark-ag/aggregated/ec2/2023_03_19_zs/zeroshot_pred_proba_50_mb.pkl",
```

+1 to using Path. This is generally something I should consider using more often rather than str for these path operations, I wasn't very familiar with them prior to you introducing me to it.

Reply:
Updated to use your recommended code snippet